src/Speckle.Sdk/Serialisation/Send/SendProcess.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Threading.Channels;
+using Open.ChannelExtensions;
+using Speckle.Sdk.Common;
+using Speckle.Sdk.Models;
+using Speckle.Sdk.Transports;
+using Speckle.Sdk.Transports.ServerUtils;
+
+namespace Speckle.Sdk.Serialisation.Send;
+
+public record SendProcessSettings(
+  int MaxSerializeThreads = 4,
+  int MaxObjectRequestSize = ServerApi.BATCH_SIZE_GET_OBJECTS,
+  int BatchWaitMilliseconds = 500
+);
+
+public sealed class SendProcess : IDisposable
+{
+  private readonly SendProcessSettings _settings = new();
+
+  public SendProcess(IModelTarget modelTarget)
+  {
+    SourceChannel = Channel.CreateUnbounded<Base>();
+    SendStage = new(modelTarget);
```

Comment:
I think this wrapping of the IDisposable is what's tripping up the CA2000 btw... shame because your logic here is sound (because SendStage is also IDisposable). CA2000 is just not particually well designed 🤦‍♀️